content/en/tracing/guide/instrument_custom_method.md

```diff
@@ -0,0 +1,177 @@
+- link: "tracing/guide/"
+  tag: ""
+  text: "All guides"
+---
+_8 minutes to complete_
+
+{{< img src="tracing/guide/custom_span/custom_span_1.png" alt="Analytics View" responsive="true" style="width:90%;">}}
+
+In order to provide you deep visibility into your business logic, Datadog APM allows you to customize the spans that make up your traces based on your needs and implementation. This empowers you to trace any method in your codebase and even specific components within methods. You can use this to optimize and monitor critical areas of your application at the granularity that works for you.
+
+Custom spans are useful when you have methods that perform multiple actions like fetching data from a payments database, sanitizing, processing and verifying the response, and finally returning the results to a customer.
+The database call and sanitization methods are already instrumented automatically, but the processing is entirely tailored to your application. By creating spans for the processing, you can optimize the timing using the APM flamegraph and monitor for errors using Datadog monitors.
+
+**Note**: When using any framework that is integrated with Datadog APM, your web requests, database queries, and cache jobs are automatically traced.
+
+## Instrument a complete method with decorators
+
+**Follow the example to get your code instrumented**.
+
+This examples walks through tracing the entire `BackupLedger.write` method to measure it's execution time and status. `BackupLedger.write` is an action that saves the current state of our transaction ledger in memory before making a call to our payments database to post a new customer charge. This happens when the `charge` endpoint of our payments service is hit:
+
+{{< img src="tracing/guide/custom_span/custom_span_2.png" alt="Analytics View" responsive="true" style="width:90%;">}}
+
+Note that the `http.request POST /charge/` span is taking a lot of time without having any direct child spans. This is a clue that this request requires further instrumentation to gain better insights into its behavior. Depending on the programming language you are using, you need to decorate your functions differently:
+
+Spans are the building blocks of APM Traces. You can instrument entire methods in your code. Follow the code sample below to create a new span that traces your method and includes it within your traces.
+
+{{< tabs >}}
+{{% tab "Java" %}}
+
+```java
+TBD - Java
+```
+
+{{% /tab %}}
+{{% tab "Python" %}}
+
+```python
+TBD - Python
+```
+
+{{% /tab %}}
+{{% tab "Ruby" %}}
+
+```ruby
+TBD - Ruby
+```
+
+{{% /tab %}}
+{{% tab "Go" %}}
+
+
+```go
+TBD - Go
+```
+
+{{% /tab %}}
+{{% tab "Node.js" %}}
+
+```javascript
+TBD - js
+```
+
+{{% /tab %}}
+{{% tab ".NET" %}}
+
+```csharp
+TBD - C#
+```
+
+{{% /tab %}}
+{{% tab "PHP" %}}
+
+```php
+TBD - PHP
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Instrument individual componenets of your methods to control the precision of your spans
```

Comment:
I'm not sure the title is really clear with "to control the precision of your spans". I think we should put as a title something that is not about "controlling the precision" but more about "have a breakdown of a custom instrumented span" (with more instrumentation). If it makes sense, can't we define this as an extra breakdown of a span?

Reply:
Updated!